packages/garbo/src/resources/skeletonOfCrimboPast.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  $familiar,
+  $item,
+  $location,
+  $phylum,
+  get,
+  have,
+  sum,
+  tuple,
+} from "libram";
+import { garboValue } from "../garboValue";
+import { globalOptions } from "../config";
+import { FamiliarMode } from "../familiar/lib";
+import { BonusEquipMode } from "../lib";
+import { appearanceRates, Location, Monster } from "kolmafia";
+
+export function knuckleboneValue(
+  mode: FamiliarMode | BonusEquipMode | Monster | Location,
+): number {
+  if (
+    get("_knuckleboneDrops", 0) >= 100 ||
+    !have($familiar`Skeleton of Crimbo Past`)
+  ) {
+    return 0;
+  }
+  const boneTradeValue = garboValue($item`knucklebone`);
+  return boneTradeValue * knuckleboneMultiplier(mode);
+}
+
+function knuckleboneMultiplier(
+  mode: FamiliarMode | BonusEquipMode | Monster | Location,
+): number {
+  if (mode === "barf") {
+    return valueLocation($location`Barf Mountain`);
+  }
+
+  if (mode === "target") {
+    return getPhylumMultiplier(globalOptions.target);
+  }
+
+  if (mode instanceof Location) return valueLocation(mode);
+  if (mode instanceof Monster) return getPhylumMultiplier(mode);
+
+  return 0;
+}
+
+function valueLocation(loc: Location): number {
+  const rates = appearanceRates(loc); // { monsterName: % }
+
+  const entries = Object.entries(rates).map(([name, rate]) =>
+    tuple(Monster.get(name), rate),
+  );
+
+  return sum(entries, ([mon, rate]) =>
+    mon !== Monster.none ? (rate / 100) * getPhylumMultiplier(mon) : 0,
```

Comment:
You don't need the Monster.none check here; Monster.none's phylum should be Phylum.none, which is handled in the `default` case of your switch.